src/keys.rs

```diff
@@ -166,14 +184,29 @@ impl NullifierDerivingKey {
     pub(crate) fn prf_nf(&self, rho: pallas::Base) -> pallas::Base {
         prf_nf(self.0, rho)
     }
+
+    /// Converts this nullifier deriving key to its serialized form.
+    pub(crate) fn to_bytes(&self) -> [u8; 32] {
+        <[u8; 32]>::from(self.0)
+    }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        let nk_bytes = <[u8; 32]>::try_from(&bytes[..32]).ok()?;
```

Comment:
Same question as above; shouldn't this fail if the length isn't exactly 32 bytes?